symphony-bdk-core/src/main/java/com/symphony/bdk/core/ServiceFactory.java

```diff
@@ -0,0 +1,61 @@
+package com.symphony.bdk.core;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.core.service.MessageService;
+import com.symphony.bdk.core.service.SessionService;
+import com.symphony.bdk.core.service.datafeed.DatafeedService;
+import com.symphony.bdk.core.service.datafeed.DatafeedVersion;
+import com.symphony.bdk.core.service.datafeed.impl.DatafeedServiceV1;
+import com.symphony.bdk.core.service.datafeed.impl.DatafeedServiceV2;
+import com.symphony.bdk.core.service.stream.StreamService;
+import com.symphony.bdk.core.service.user.UserService;
+import com.symphony.bdk.gen.api.DatafeedApi;
+import com.symphony.bdk.gen.api.MessagesApi;
+import com.symphony.bdk.gen.api.SessionApi;
+import com.symphony.bdk.gen.api.StreamsApi;
+import com.symphony.bdk.gen.api.UserApi;
+import com.symphony.bdk.gen.api.UsersApi;
+
+class ServiceFactory {
+
+  private final UserService userService;
```

Comment:
Why does this class should contain single instance for each service? Is it really a factory?

Reply:
I agree that it's not really a factory. I want to decouple the service stuff in SymphonyBdk to a dedicated class. The instance of each class is initialized at the constructor given an `AuthSession`. And I think for one `AuthSession`, we should only one instance of each service.
Maybe I should rename this class?